src/languages/swift.js

```diff
@@ -56,7 +60,7 @@ export default function(hljs) {
   // find all the regular keywords
   const KEYWORDS = {
     $pattern: either(
-      /\b\w+(\(\w+\))?/, // kw or kw(arg)
```

Comment:
Ah your problem with this was your tuple rule eating the `()`, yes?  That's why we try to avoid parsing and do simpler rules... if your rule only matched `blah:` and avoided trying to figure out for *certain* that something looked like a tuple you'd avoid this problem completely.  Since you're only using it as a guard and not highlighting it's not the end of the world if it matches a bit wide (as long as it's not stealing things we need to highlight).

Are there other syntactic things in Swift that use `ident:`?

Reply:
I'm also guessing it's technically wrong to assume all `()` combos would be tuples... so approach is 100% "correct" in the Swift language sense.

Reply:
See above, there's just a looooot of things that use `identifier:` in Swift. Within parentheses, it's almost always a tuple element/function parameter name. I'll look into it more and see what cases I missed that may cause issues.

Of course, parentheses can also be just parentheses in expressions, but because tuple elements are also expressions, that should match nicely.

Reply:
> See above, there's just a looooot of things that use identifier: in Swift. 

_But if we're not highlighting those things, it's irrelevant how many variant there are..._ and if we are it still should mostly be irrelevant since we'd likely be highlighting within a child scope and the parent scope (which had the simpler `blah:` rule) would no longer be in effect.  And if we were highlighting them ALL in the parent scope then this problem goes away entirely. 

But _right now_ I don't see any need for the tuple rule.  And simplifying it also simplifies keywords.

Is all this just to prevent keywords being highlighted in tuples?  Wouldn't a negative lookahead for `:` on keywords `$pattern` solve this?